### PR TITLE
Adding testing framework to CI and CMake

### PR DESCRIPTION
**Pull request by q349wang**
image

image

Look how nice this is :D

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,6 +94,13 @@ target_include_directories(antioch-base-test PUBLIC
...
+  gtest
 )
+
+include(GoogleTest)
+# Windows environment on GitHub Actions is quirky
+if (DEFINED ENV{GITHUB_ACTIONS} AND WIN32)
+gtest_add_tests(TARGET antioch-base-test)
+else()
+gtest_discover_tests(antioch-base-test)
+endif()
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt]: nl

File: CMakeLists.txt [hunk 2]
```diff
@@ -91,6 +94,13 @@ target_include_directories(antioch-base-test PUBLIC
 )
 target_link_libraries(antioch-base-test PUBLIC
   antioch-base
-  GTest::gtest_main
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt:94]: is this necessary? GTest::** is the recommendation from Google

File: .github/workflows/build-and-test-antioch.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            run: cmake --build build
+
+    build-on-linux:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout Repo
+            uses: actions/checkout@v3
+
+          - name: Get CMake
+            uses: lukka/get-cmake@latest
```
Comment (Victor-C-Zhang) [inline .github/workflows/build-and-test-antioch.yml:31]: is this necessary?

File: .github/workflows/build-and-test-antioch.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+          - run: mkdir build
+          - run: cd build
```
Comment (Victor-C-Zhang) [inline .github/workflows/build-and-test-antioch.yml]: combine or use yaml multiline
```
- run: |
  mkdir build
  cd build
```

File: CMakeLists.txt (hunk 2 above)
Comment (q349wang) [reply CMakeLists.txt:94]: Nah I can do that instead

File: .github/workflows/build-and-test-antioch.yml (hunk 3 above)
Comment (q349wang) [reply .github/workflows/build-and-test-antioch.yml:31]: Yeah I believe so it needs to get cmake

Comment (Victor-C-Zhang) [reply .github/workflows/build-and-test-antioch.yml:31]: take a look at the github actions i wrote in GraphStreamingProject/GraphZeppelin. IIRC we don’t need to get cmake for ubuntu

**Comment by Victor-C-Zhang** [comment]
Approving pending investigating potentially not using get cmake on ubuntu

**Review by Victor-C-Zhang — APPROVED** [review]

**Pull request merged**